homeassistant/components/iron_os/config_flow.py

```diff
@@ -36,30 +41,62 @@ async def async_step_bluetooth_confirm(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Confirm discovery."""
+
+        errors: dict[str, str] = {}
+
         assert self._discovery_info is not None
         discovery_info = self._discovery_info
         title = discovery_info.name
 
         if user_input is not None:
-            return self.async_create_entry(title=title, data={})
+            device = Pynecil(discovery_info.address)
+            try:
+                await device.connect()
+            except (CommunicationError, BleakError, TimeoutError):
+                _LOGGER.debug("Cannot connect:", exc_info=True)
+                errors["base"] = "cannot_connect"
+            except Exception:
+                _LOGGER.exception("Unexpected exception:")
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(title=title, data={})
+            finally:
+                await device.disconnect()
 
         self._set_confirm_only()
         placeholders = {"name": title}
         self.context["title_placeholders"] = placeholders
         return self.async_show_form(
-            step_id="bluetooth_confirm", description_placeholders=placeholders
+            step_id="bluetooth_confirm",
+            description_placeholders=placeholders,
+            errors=errors,
         )
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Handle the user step to pick discovered device."""
+
+        errors: dict[str, str] = {}
+
         if user_input is not None:
             address = user_input[CONF_ADDRESS]
             title = self._discovered_devices[address]
-            await self.async_set_unique_id(address, raise_on_progress=False)
-            self._abort_if_unique_id_configured()
-            return self.async_create_entry(title=title, data={})
+            device = Pynecil(address)
+            try:
+                await device.connect()
+            except (CommunicationError, BleakError, TimeoutError):
+                _LOGGER.debug("Cannot connect:", exc_info=True)
+                errors["base"] = "cannot_connect"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(address, raise_on_progress=False)
+                self._abort_if_unique_id_configured()
```

Comment:
Would it be possible to do this before the connect? I don't think you need to connect (and disconnect) to know it already exists since the address won't change.